### PR TITLE
Add Dask Integration: vineyard as the data source for dask

### PR DESCRIPTION
**Pull request by andydiwenzhu**
Signed-off-by: Diwen Zhu <diwen.zdw@alibaba-inc.com>

What do these changes do?
-------------------------

Add two resolvers for dask: GlobalTensor --> dask.Array, GlobalDataFrame --> dask.dataframe

Related issue number
--------------------

Part of #412.

### CODE REVIEW

File: python/vineyard/contrib/ml/distributed/dask.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+def dask_dataframe_builder(client, value, builder, **kw):
+    # TODO: build dask.dataframe to vineyard::GlobalDataFrame
+    pass
+
+
+def dask_array_resolver(obj, resolver, **kw):
+    def get_partition(socket, obj_id):
+        client = vineyard.connect(socket)
+        np_value = client.get(obj_id)
+        return da.from_array(np_value)
```
Comment (sighingnow) [inline python/vineyard/contrib/ml/distributed/dask.py:43]: Is there a copy? (I'm not sure.

**Review by sighingnow — COMMENTED** [review]
Could we have a package called vineyard-dask? under vineyard/contrib/dask.

It is quite strange to put it under ml/distributed....

**Comment by sighingnow** [comment]
Seems the resolver assumes both the tensor and dataframe are chunked along the `x` ax,  it not alway be true, as the tensor/dataframe may come from mars.

We haven't put enough efforts in other ML integrations, I know it is not trivial, but it does need to be fixed. As the dask already natively supports split chunks along both the `x` and `y` axis, and we have already record enough information in the metadata, could we do right thing here for dask? Otherwise it would be depreacated and replaced by a new implementation someday.

**Comment by andydiwenzhu** [comment]
> Seems the resolver assumes both the tensor and dataframe are chunked along the `x` ax, it not alway be true, as the tensor/dataframe may come from mars.
> 
> We haven't put enough efforts in other ML integrations, I know it is not trivial, but it does need to be fixed. As the dask already natively supports split chunks along both the `x` and `y` axis, and we have already record enough information in the metadata, could we do right thing here for dask? Otherwise it would be depreacated and replaced by a new implementation someday.

**Comment by sighingnow** [comment]
Andy @andydiwenzhu  I didn't mean closing the issue. We could

1. put those things into a vineyard-dask package
2. leave a check to raise an `NotImplementedError` exception when the source is not chunked as what we want

Then we could get this pull request landed.

**Comment by andydiwenzhu** [comment]
> Andy @andydiwenzhu I didn't mean closing the issue. We could
> 
> 1. put those things into a vineyard-dask package
> 2. leave a check to raise an `NotImplementedError` exception when the source is not chunked as what we want
> 
> Then we could get this pull request landed.

OK. That was a bad click, don't worry :)

**Review by sighingnow — APPROVED** [review]

**Pull request merged**